src/deal-tracking/DealTrackingService.ts

```diff
@@ -31,19 +30,110 @@ export default class DealTrackingService extends BaseService {
       const client = clientState.stateKey;
       const lastDeal = await Datastore.DealStateModel.find({ client }).sort({ dealId: -1 }).limit(1);
       try {
-        await this.insertDealFromFilfox(client, lastDeal.length > 0 ? lastDeal[0].dealId : 0);
+        await this.insertDealFromFilscan(client, lastDeal.length > 0 ? lastDeal[0].dealId! : 0);
       } catch (error) {
-        this.logger.error('Encountered an error when importing deals from filfox', { error });
+        this.logger.error('Encountered an error when importing deals from filescan', error);
       }
       try {
         await this.updateDealFromLotus(client);
       } catch (error) {
-        this.logger.error('Encountered an error when updating deals from lotus', { error });
+        this.logger.error('Encountered an error when updating deals from lotus', error);
       }
     }
   }
 
-  private async insertDealFromFilfox (client: string, lastDeal: number): Promise<void> {
+  /**
+   * Read from filscan api for PublishStorageDeal status
+   *
+   * @param client
+   * @param lastDeal
+   */
+  private async insertDealFromFilscan (client: string, lastDeal: number): Promise<void> {
+    this.logger.debug('updating deals from filscan', { client, lastDeal });
+    let page = 0;
+    let response;
+    do {
+      const offset = page * 25;
+      let breakOuter = false;
+
+      let url = 'https://api.filscan.io:8700/rpc/v1';
+      if (client.startsWith('t')) {
+        url = 'https://calibration.filscan.io:8700/rpc/v1';
+      }
+      this.logger.debug(`Fetching from ${url}`);
+      response = await axios.post(url, {
+        id: 1,
+        jsonrpc: '2.0',
+        params: [client, offset, 25],
+        method: 'filscan.GetMarketDeal'
+      }, {
+        headers: {
+          'content-type': 'application/json'
+        }
+      });
+      if (Array.isArray(response.data['result']['deals'])) {
+        const jsonResult = response.data['result'];
+        this.logger.debug(`Received ${offset} out of ${jsonResult['total']} deal entries.`);
+        for (const deal of jsonResult['deals']) {
+          if (deal['dealid'] <= lastDeal || jsonResult['deals'].length < 25) {
+            breakOuter = true;
+            this.logger.info('Reached to the last page or last checked deal.');
+            break;
+          }
+          const proposedDeal = await Datastore.DealStateModel.findOne({
```

Comment:
Add index for the field pieceCid and dealId in Datastore.ts (which I think should be enough for those queries)

Reply:
It's possible that filscan can return the same entry twice. For example, on the first page and then on the next page since new messages of new block gets ingested. If this happens and when there appears two entries that match the query, the second one will be updated to the same dealId. 
To avoid this, you may need to find one by deal id first, if it already exists, then skip.